### PR TITLE
Manually register missing UDFs

### PR DESCRIPTION
**Pull request by aastha25**
Manually registering several more UDFs. 

Testing Done:
Built successfully. Spotless check successful. 
Ran with views that were previously failing. 

Also fixed the return type for previously added UDF IsValidKeyUDF.

### CODE REVIEW

File: coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java [hunk 1]
```diff
@@ -41,10 +41,12 @@ private FunctionReturnTypes() {
     return opBinding.isOperandLiteral(1, false) ? type2 : type1;
   };
 
+  public static final SqlReturnTypeInference BOOLEAN = ReturnTypes.explicit(SqlTypeName.BOOLEAN);
   public static final SqlReturnTypeInference STRING = ReturnTypes.explicit(SqlTypeName.VARCHAR);
   public static final SqlReturnTypeInference BINARY = ReturnTypes.explicit(SqlTypeName.BINARY);
   public static final SqlReturnTypeInference BIGINT = ReturnTypes.explicit(SqlTypeName.BIGINT);
   public static final SqlReturnTypeInference SMALLINT = ReturnTypes.explicit(SqlTypeName.SMALLINT);
+  public static final SqlReturnTypeInference INTEGER = ReturnTypes.explicit(SqlTypeName.INTEGER);
```
Comment (ljfgem) [inline coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java:49]: May just use `ReturnTypes.BOOLEAN` and `ReturnTypes.INTEGER` provided by calcite directly, so we don't need to add these 2 types here.
Actually I found `DATE` and `BIGINT` are also defined in `org.apache.calcite.sql.type.ReturnTypes`, could you help to remove them from this class too?

**Review by ljfgem — COMMENTED** [review]
Thanks @aastha25 for this PR, left one nit.

File: coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java (hunk 1 above)
Comment (aastha25) [reply coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java]: Thanks for suggesting it. Updated the references. 
Also rebased.

**Review by ljfgem — APPROVED** [review]
LGTM, thanks @aastha25!

**Pull request merged**